src/main/java/org/jabref/logic/importer/CitationBasedParserFetcher.java

```diff
@@ -0,0 +1,111 @@
+import kong.unirest.json.JSONException;
+import kong.unirest.json.JSONObject;
+
+/**
+ * Provides a convenient interface for citation-based fetcher, which follow the usual four-step procedure:
+ * 1. Send request with doi of the entry
+ * 2. Parse the response to get a list of DOI's
+ * 3. Send new request to get bibliographic information for every DOI
+ * 4. Parse the response and create a List of {@link BibEntry}
+ */
+public interface CitationBasedParserFetcher extends CitationFetcher {
+
+    /**
+     * Constructs a URL-String based on the given DOI's.
+     *
+     * @param dois the dois to look information for
+     */
+    String getURLForEntries(String... dois);
+
+    default List<BibEntry> searchCitedBy(BibEntry entry) throws FetcherException {
+        Objects.requireNonNull(entry);
+        return performSearch(entry, OpenCitationFetcher.SearchType.CITEDBY);
+    }
+
+    default List<BibEntry> searchCiting(BibEntry entry) throws FetcherException {
+        Objects.requireNonNull(entry);
+        return performSearch(entry, OpenCitationFetcher.SearchType.CITING);
+    }
+
+    /**
+     * Searches specified URL for related articles based on a {@link BibEntry}
+     * and the citation type {@link org.jabref.logic.importer.fetcher.OpenCitationFetcher.SearchType}
+     *
+     * @param entry      entry to search bibliographic information for
+     * @param searchType type of search to perform (CITING, CITEDBY)
+     * @return a list of {@link BibEntry}, which are matched by the query (may be empty)
+     * @throws FetcherException Error message passed to {@link org.jabref.gui.entryeditor.CitationRelationsTab}
+     */
+    default List<BibEntry> performSearch(BibEntry entry, OpenCitationFetcher.SearchType searchType) throws FetcherException {
+        Objects.requireNonNull(entry);
+        Objects.requireNonNull(searchType);
+        String doi = entry.getField(StandardField.DOI).orElse("");
+
+        List<BibEntry> list = new ArrayList<>();
+        try {
+            JSONArray json = Objects.requireNonNull(readJsonFromUrl(getURLForEntries(doi)));
+            if (json.isEmpty()) {
+                return list;
+            }
+            String[] items = json.getJSONObject(0).getString(searchType.label).split("; ");
+            if (!Arrays.equals(items, new String[] {""})) {
+                JSONArray metaArray = Objects.requireNonNull(readJsonFromUrl(getURLForEntries(items)));
+                if (metaArray.isEmpty()) {
+                    return list;
+                }
+                for (int i = 0; i < metaArray.length(); i++) {
+                    list.add(createNewEntry(metaArray.getJSONObject(i)));
+                }
+            }
+        } catch (IOException | JSONException | NullPointerException e) {
+            throw new FetcherException("Couldn't connect to API! Please try again.");
+        }
+        return list;
+    }
+
+    default BibEntry createNewEntry(JSONObject jsonObject) {
```

Comment:
What is the purpose of a default method that returns null?
A default method in an interface should provide a valid default implementation like in an abstract class.
In addtion it's never a good idea to return null. For such a case Optional has been introduced